### ISSUE TITLE
Bgziped VCF is illegal.

### ISSUE THREAD

**Issue by niyarin** [chrovis/cljam#251]
We can reproduce it with the following code.
```clojure
(with-open [wtr (vcf/writer "./test.vcf.gz"
                            {:format nil :contig {:id "1", :length nil} :filter [{:id "PASS", :description "All filters passed"}] :info nil}
                            ["CHROM" "POS" "ID" "REF" "ALT" "QUAL" "FILTER" "INFO" "FORMAT"])]
  (vcf/write-variants wtr [{:chr "1" :pos 10 :ref "A" :alt ["G"]}]))

(with-open [rdr (vcf/reader "./test.vcf.gz")]
  (vcf/read-variants rdr))
;Error printing return value (IOException) at java.io.BufferedReader/ensureOpen (BufferedReader.java:122).
```
The generated bgzip file also fails with the'bcftools index'.
Now, if we do `bgzip -d` and` bgzip`, the error doesn't occur.

**Comment by alumi** [comment]
Hi @niyarin, thanks for filing the issue! 👍 

I followed the procedure you posted. It required some corrections but I confirmed that the problem is actually reproduced.

1. we have to realize the variants before exiting from the `with-open` macro.

    ```clojure
    (with-open [rdr (vcf/reader "./test.vcf.gz")]
      (doall (vcf/read-variants rdr)))
    ```

2. `:contig` must be a sequence

    ```clojure
    :contig [{:id "1", :length nil}]
    ```

3. `FORMAT` is followed by samples

   ```clojure
   ["CHROM" "POS" "ID" "REF" "ALT" "QUAL" "FILTER" "INFO" "FORMAT" "SAMPLE"]
   ```

The header of the written vcf.gz file was `1f 8b 08 00 00 00 00 00 00 ff 2d 8c d1 0a 82 30 ...` which means that this is a GZIP file but not a BGZIP file.
The root cause is `vcf-writer` uses `compressor-output-stream` and it doesn't return a `BGZFOutputStream`.
https://github.com/chrovis/cljam/blob/23392cbec4960ddf3da4c7653d0823f5177c1e7f/src/cljam/io/vcf.clj#L145
https://github.com/chrovis/cljam/blob/23392cbec4960ddf3da4c7653d0823f5177c1e7f/src/cljam/util.clj#L127-L128

`compressor-output-stream` is used by VCF, BED, WIG, FASTQ, FASTA and GFF.
I think we can safely change the output format from GZIP to BGZIP.
